components/AssetPage/Download/Download.tsx

```diff
@@ -232,6 +232,15 @@ const Download = ({ assetID, data, files, setPreview, patron }) => {
         }
       }
     }
+
+    // Add the preview image to the zip
+    if (files.preview) dlFiles.push({
+      url: files.preview,
+      path: `${assetID}_preview.png`
+      // TODO: is the size of this image stored in the db?
```

Comment:
No it's not. I believe it might be necessary to have the final bundle size known before downloading so this might be a roadblock.

Reply:
Ok, so it's gonna be a tough one. For now, I hardcoded `0` for the size of the preview images, which is obviously a lie, but it's good for testing. 

From what I see we have two problems here:
1. we need to get the file sizes of all the thumbnail images
2. we need to get that data into Firebase (if that's what you're still using)

For _issue 1_ I can use ` https://api.polyhaven.com/assets` to get the list of all the assets, and then write a bash script that downloads them and uses `exiftool` to get their sizes. I can save this data to a `csv`/`json`.

As for _issue 2_, I don't have experience specifically with Firebase, but I believe we can write another script (that relies on `firebase-tools` (https://www.npmjs.com/package/firebase-tools)) to save the data in the db. This second script would have to run on your computer, because I have no access to Firebase. If you're using windows, then we might have to use Node.js instead of bash for this second script. 

What do you think?

Reply:
Okay, I came up with a hacky approach: https://github.com/Poly-Haven/polyhaven.com/pull/55/files#r1018417712